central/administration/events/datastore/internal/writer/writer_impl.go

```diff
@@ -37,16 +40,41 @@ func (c *writerImpl) resetNoLock() {
 }
 
 func (c *writerImpl) flushNoLock(ctx context.Context) error {
-	eventChunk := make([]*storage.AdministrationEvent, 0, len(c.buffer))
-	for _, event := range c.buffer {
-		eventChunk = append(eventChunk, event)
+	eventsToAdd := sliceutils.ShallowClone(maputil.Values(c.buffer))
+
+	// Short-circuit here in case we have no events to be added.
```

Comment:
Can't we short circuit already first thing first in this function if `len(c.buffer) == 0`?